pipelines/zendesk/zendesk.py

```diff
@@ -0,0 +1,184 @@
+import dlt
+from dlt.common import logger
+from dlt.common.typing import TDataItem
+from dlt.extract.source import DltResource
+from pendulum import DateTime, datetime, now
+from pipelines.zendesk.helpers.api_helpers import ZendeskCredentials, auth_zendesk, basic_load, process_ticket
+from typing import Iterator, Sequence, Dict
+from zenpy import Zenpy
+
+
+EXTRA_RESOURCES = [
+    "activities", "automations", "custom_agent_roles", "dynamic_content", "group_memberships", "job_status", "macros", "organization_fields", "organization_memberships", "recipient_addresses",
+    "requests", "satisfaction_ratings", "sharing_agreements", "skips", "suspended_tickets", "targets", "ticket_forms", "ticket_metrics", "triggers", "user_fields", "views", "tags"
+]
+resources_to_be_loaded = ["users", "sla_policies", "groups", "organizations", "brands"]
+
+CURRENT_YEAR = now().year
+FIRST_DAY_OF_CURRENT_YEAR = datetime(year=CURRENT_YEAR, month=1, day=1)
+FIRST_DAY_OF_MILLENNIUM = datetime(year=2000, month=1, day=1)
+
+
+@dlt.source
+def zendesk_chat(credentials: Dict[str, str]) -> DltResource:
+    """
+    The source for the dlt pipeline. It returns all the basic information.
+    @:param credentials: read as a dict, as filled in .dlt.secrets.toml
+    @:returns: multiple dlt resources
+    """
+    # Authenticate
+    zen_credentials = ZendeskCredentials(credentials)
+    zendesk_client = auth_zendesk(credentials=zen_credentials)
+    return chats_table(zendesk_client=zendesk_client)
+
+
+@dlt.source
+def zendesk(credentials: Dict[str, str] = dlt.secrets.value, load_all: bool = True,
```

Comment:
Please use `ZendeskCredentials` directly. Here you have examples (made for this case specifically). It will handle the auth  the options for you automatically

https://github.com/dlt-hub/dlt/blob/0.2.0a23/docs/technical/secrets_and_config.md#working-with-alternatives-of-credentials-union-types